### PR TITLE
feat(experimental): add OEE function

### PR DESCRIPTION
**Pull request by alespour**
This PR implements OEE function per https://github.com/influxdata/flux/issues/3479.

The package provides `computeAPQ` and pipeline `APQ` functions. They require that production events stream tables have `state` (production state, eg. `"running"`, `"stopped"`) and part events input `partCount` (integer) and `badCount` (integer) columns.

### Done checklist
- [x] Test cases written

### CODE REVIEW

**Comment by alespour** [comment]
@nathanielc can you please have a look at this? thank you.

File: stdlib/experimental/oee/oee.flux [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    runningState,
+    plannedTime,
+    idealCycleTime
+) => {
+    availability = productionEvents
+        |> events.duration(unit: 1ns, columnName: "runTime")
+        |> filter(fn: (r) => r.state == runningState)
+        |> sum(column: "runTime")
+        |> duplicate(column: "_stop", as: "_time")
+        |> map(fn: (r) => ({ r with availability: float(v: r.runTime) / float(v: int(v: plannedTime)) }))
```
Comment (nathanielc) [inline stdlib/experimental/oee/oee.flux:18]: small nit, you can do the `duplicate` step as part of the `map` step to make it a bit faster. 
```
|> map(fn: (r) => ({ r with _time: r._stop, availability: float(v: r.runTime) / float(v: int(v: plannedTime)) }))
```

File: stdlib/experimental/oee/oee.flux [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package oee
+
+import "contrib/tomhollingworth/events"
+import "experimental"
+
+computeAPQ = (
```
Comment (nathanielc) [inline stdlib/experimental/oee/oee.flux:6]: Can you add a descriptive comment for what each argument is?

File: stdlib/experimental/oee/oee.flux [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            quality: (float(v: right.partCount) - float(v:left.badCount)) / float(v: right.partCount)
+    }))
+
+    return experimental.join(left:performance, right: quality, fn: (left, right) => ({left with
+        quality: right.quality,
+        oee: left.availability * left.performance * right.quality
+    }))
+}
+
+APQ = (
```
Comment (nathanielc) [inline stdlib/experimental/oee/oee.flux:40]: Same here can you document how this function should be used in a comment?

**Review by nathanielc — COMMENTED** [review]
Overall looks great, just a few things to update.

**Review by scbrickley — COMMENTED** [review]
Apart from the points already made by @nathanielc, nothing here sticks out to me as needing edits.

**Comment by alespour** [comment]
@nathanielc I have checked the linked issue once again and now I'm not sure this PR meets the expectations.

The counts in sample data in #3479 are cumulative, so the `APQ` uses `difference`  to get the number of parts. But, in the proposal of higher level function that takes productions and parts events stream on input, the `partEvents` is described as a stream where each row represents production of a single part. So the current implementation in this PR is wrong I think... depending on whether counts are cumulative or not, the `APQ` function should either use `difference` or `sum`, I guess?

**Comment by nathanielc** [comment]
The counts are cumulative so using difference is the correct approach. I just verified based on the description we got from some potential users that they are tracking cumulative counts. The counts can reset so it should be a `nonNegative` difference, but not a `sum`.

**Comment by alespour** [comment]
So this PR should be ok as it is. Thank you for the clarification.

**Review by nathanielc — APPROVED** [review]

**Pull request merged**